CMakeLists.txt

```diff
@@ -187,11 +187,21 @@ set(SOURCE_FILES_CPP_WRAPPER
         include/snowflake/Param.hpp
         include/snowflake/Exceptions.hpp
         include/snowflake/jwtWrapper.h
+        include/snowflake/result_set_arrow.h
+        include/snowflake/result_set_json.h
+        include/snowflake/ResultSet.hpp
+        include/snowflake/ResultSetArrow.hpp
+        include/snowflake/ResultSetJson.hpp
```

Comment:
Does a C application compiling against libsnowflakeclient need these header files to compile? 
The resultset is something internal and need not be exposed to outside world.

Reply:
include/snowflake/result_set_arrow.h
        include/snowflake/result_set_json.h
Files do not exist

Reply:
This is something that was raised during internal review as well.
I've moved these header files to `lib/` and `cpp/lib/` depending on the language used, but it seems like I somehow missed the files in `lib/`. I've amended this.